### PR TITLE
Use gh-issues-inspector to improve error reporting 

### PR DESCRIPTION
**Pull request by orta**
Fixes: https://github.com/CocoaPods/CocoaPods/issues/4817

Adds GitHub issue search to our exception handler, this is mainly to make it easier for people to find existing issues rather than writing new ones.

### CODE REVIEW

File: lib/cocoapods/command/install.rb [hunk 1]
```diff
@@ -30,11 +30,22 @@ def self.options
...
+        begin
+          verify_podfile_exists!
+          installer = installer_for_config
+          installer.repo_update = repo_update?(:default => false)
+          installer.update = false
+          installer.install!
+
+        rescue Exception => e
+          whitelist_exceptions = [Pod::Informative]
+          unless whitelist_exceptions.include? e.class
```
Comment (orta) [inline lib/cocoapods/command/install.rb]: What other errors would we _not_ want issue checks for?

File: lib/cocoapods/user_interface.rb [hunk 2]
```diff
@@ -447,3 +447,72 @@ def self.warn(message)
...
+        when 82_801..172_000 then 'a day ago' # 86400 = 1 day
+        when 172_001..518_400 then ((a + 800) / (60 * 60 * 24)).to_i.to_s + ' days ago'
+        when 518_400..1_036_800 then 'a week ago'
+        when 1_036_801..4_147_204 then ((a + 180_000) / (60 * 60 * 24 * 7)).to_i.to_s + ' weeks ago'
+        else strftime("%d %b %Y")
+        end
+      end
+    end
+  end
+end
```
Comment (orta) [inline lib/cocoapods/user_interface.rb]: this is almost a verbatim version of [my version](https://github.com/orta/gh-issues-inspector/blob/master/lib/evidence.rb) in the inspector itself ( given that I made it with CP in mind )

File: lib/cocoapods/command/install.rb [hunk 3]
```diff
@@ -30,11 +30,22 @@ def self.options
...
-        installer.update = false
-        installer.install!
+        begin
+          verify_podfile_exists!
+          installer = installer_for_config
+          installer.repo_update = repo_update?(:default => false)
+          installer.update = false
+          installer.install!
+
+        rescue Exception => e
```
Comment (segiddins) [inline lib/cocoapods/command/install.rb]: this should be part of the error report, imho

File: lib/cocoapods/user_interface.rb [hunk 4]
```diff
@@ -447,3 +447,72 @@ def self.warn(message)
     end
   end
 end
+
+#---------------------------------------------------------------------------#
+
+module Pod
+  module GHIssuesInspectorDelegate
```
Comment (segiddins) [inline lib/cocoapods/user_interface.rb]: put this in a new file, namespaced under `UserInterface`?

**Comment by orta** [comment]
Cool, I agreed with all feedback, it feels a lot better now

File: spec/functional/command/install_spec.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 module Pod
   describe Command::Install do
     it 'tells the user that no Podfile or podspec was found in the project dir' do
-      exception = lambda { run_command('install', '--no-repo-update') }.should.raise Informative
+      exception = lambda { run_command('install') }.should.raise Informative
```
Comment (orta) [inline spec/functional/command/install_spec.rb]: that arg doesn't exist anymore `/shrug`

File: spec/unit/user_interface/error_report_spec.rb [hunk 6]
```diff
@@ -110,6 +110,37 @@ def remove_color(string)
         message = remove_color(message)
         message.should == '[!] at -'
       end
+    
+      it "handles inspector_successfully_recieved_report" do
+        time = Time.new(2016, 05, 13)
+        Time.stubs(:now).returns(time)
+        
+        url = 'https://api.github.com/search/issues?q=Testing%252Brepo%253Acocoapods%252Fcocoapods&sort=created&order=asc'
```
Comment (segiddins) [inline spec/unit/user_interface/error_report_spec.rb]: think you need to remove the URL escapes from this

File: spec/unit/user_interface/inspector_reporter_spec.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    it "handles inspector_started_query" do
+      inspector = Inspector::Inspector.new "cocoapods", "cocoapods"
+      reporter = UserInterface::InspectorReporter.new
+      reporter.inspector_started_query("query", inspector)
+        
+      UI.output.should.match /Looking for related issues on cocoapods\/cocoapods/
+    end
+    
+    it "handles inspector_successfully_recieved_report" do
+      url = 'https://api.github.com/search/issues?q=Testing%252Brepo%253Acocoapods%252Fcocoapods&sort=created&order=asc'
```
Comment (segiddins) [inline spec/unit/user_interface/inspector_reporter_spec.rb]: think you need to remove the URL escapes here

Comment (orta) [inline spec/unit/user_interface/inspector_reporter_spec.rb]: Yeah, definitely, there's a bug with the inspector around this 👍

File: lib/cocoapods/user_interface/inspector_reporter.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+      def print_issue_full(issue)
+        UI.puts " - #{issue.title}"
+        UI.puts "   #{issue.html_url} [#{issue.state}] [#{issue.comments} comment#{issue.comments == 1 ? '' : 's'}]"
+        UI.puts "   #{pretty_date(issue.updated_at)}"
+        UI.puts ""
+      end
+      
+      # Taken from http://stackoverflow.com/questions/195740/how-do-you-do-relative-time-in-rails
+      def pretty_date(date_string)
```
Comment (KrauseFx) [inline lib/cocoapods/user_interface/inspector_reporter.rb]: We already have that in the gem itself, why do we have it here again?

File: spec/unit/user_interface/error_report_spec.rb (hunk 6 above)
Comment (KrauseFx) [inline spec/unit/user_interface/error_report_spec.rb]: Yeah, basically https://github.com/orta/gh-issues-inspector/pull/7

File: spec/unit/user_interface/inspector_reporter_spec.rb (hunk 7 above)
Comment (KrauseFx) [inline spec/unit/user_interface/inspector_reporter_spec.rb]: Replied via https://github.com/CocoaPods/CocoaPods/pull/5421/files#r67609116

**Comment by KrauseFx** [comment]
For reference, this requires some changes after the rename https://github.com/fastlane/fastlane/pull/5162/commits/ac961b3a82b977eecb0a33b00d9e0f5d60313b14

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! \nYou can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
    
    
  </tbody>
</table>
  

  

  



Here's an example of your CHANGELOG entry:

```markdown
* [WIP] Use gh-issues-inspector to improve error reporting  
  [orta](https://github.com/orta)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by segiddins** [comment]
@orta mind if I rebase this & get rubocop passing?

**Comment by orta** [comment]
Yep - sure! - it was taking a few hours to do CI each time so I couldn't seem to get it to pass too quickly

**Comment by segiddins** [comment]
👍

**Comment by orta** [comment]
OK, I think that's everything rebased and triple-checked the rubocop

**Comment by orta** [comment]
Anyone know why this is failing? Only occurred when rebasing

File: Gemfile.lock [hunk 9]
```diff
@@ -108,7 +108,9 @@ PATH
       colored (~> 1.2)
       escape (~> 0.0.4)
       fourflusher (~> 1.0.1)
+      gh_inspector (~> 1.0)
       molinillo (~> 0.5.0)
+      gh-issues-inspector (~> 0.5.0)
```
Comment (segiddins) [inline Gemfile.lock]: this line shouldn't be here...

**Comment by orta** [comment]
Aye, good catch, amended the commit 👍

**Comment by orta** [comment]
Interesting, I've never heard of Inch, however - I have now added a lot of inline docs to satisfy it 👯

File: CHANGELOG.md [hunk 10]
```diff
@@ -31,6 +31,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 * Accept `pod repo push` with URL instead of only repo name  
   [Mark Schall](https://github.com/maschall)
   [#5572](https://github.com/CocoaPods/CocoaPods/pull/5572)
+* Shows GitHub Issues that could be related to potential issues.
```
Comment (segiddins) [inline CHANGELOG.md]: missing a blank line above

Comment (orta) [inline CHANGELOG.md]: amended - thanks

File: Gemfile.lock (hunk 9 above)
Comment (segiddins) [inline Gemfile.lock]: this still needs to be addressed

**Comment by orta** [comment]
Doh, re-refixed 💃

**Comment by orta** [comment]
Wooo - it's merged!

**Comment by KrauseFx** [comment]
This is great :rocket:

**Pull request merged**